Update sigstore-go in gh CLI to v0.5.1

steiza opened the pull request:
The v0.5.x sigstore-go release included some breaking changes, so we are proactively making pull requests to update the dependency with the associated code changes.

<!--
  Thank you for contributing to GitHub CLI!
  To reference an open issue, please write this in your description: `Fixes #NUMBER`
-->

steiza committed:
Update sigstore-go in gh CLI to v0.5.1

The v0.5.x sigstore-go release included some breaking changes, so we are
proactively making pull requests to update the dependency with the
associated code changes.

Signed-off-by: Zach Steindler <steiza@github.com>

cliAutomation commented:
Hi! Thanks for the pull request. Please ensure that this change is linked to an issue by mentioning an issue number in the description of the pull request. If this pull request would close the issue, please put the word 'Fixes' before the issue number somewhere in the pull request body. If this is a tiny change like fixing a typo, feel free to ignore this message.

steiza committed:
Run go fmt

Signed-off-by: Zach Steindler <steiza@github.com>

williammartin commented:
I don't have an obvious reason for the linting failure but probably worth trying bumping the golangci-lint version here: https://github.com/cli/cli/blob/trunk/.github/workflows/lint.yml#L35

I _suspect_ the version of golangci lint we use in ci was compiled with an older version of go than the new toolchain requires. It's just a hunch though.

steiza committed:
Update integration tests and golangci-lint version

Signed-off-by: Zach Steindler <steiza@github.com>

steiza committed:
Update internal/codespaces/rpc/invoker.go

Based on the guidance on
https://github.com/grpc/grpc-go/blob/master/Documentation/anti-patterns.md

Signed-off-by: Zach Steindler <steiza@github.com>

steiza commented:
CC @josebalius for the changes to `internal/codespaces/rpc/invoker.go` to get linting to pass.

williammartin commented on go.mod:
```diff
@@ -1,14 +1,14 @@
 module github.com/cli/cli/v2
 
-go 1.22
+go 1.22.0
```
It's quite irritating that we are being forced to change this because `sigstore-go` was forced to, because `sigstore` was forced to because `letsencrypt/boulder` unnecessarily used `1.22.0` instead of `1.22` to get `range-over-int` (at least, that appears to be the chain). Now _our_ consumers (if we have any) are forced to bump as well.

Unfortunately I don't see any way around it right now. Perhaps we can communicate upstream for when they change in future.

williammartin commented on internal/codespaces/rpc/invoker.go:
```diff
@@ -116,9 +116,8 @@ func connect(ctx context.Context, fwd portforwarder.PortForwarder) (Invoker, err
-			grpc.WithBlock(),
 		}
-		conn, err = grpc.DialContext(connectctx, localAddress, opts...)
+		conn, err = grpc.NewClient(localAddress, opts...)
```
Looks like a legit change as described by:
 * https://github.com/grpc/grpc-go/issues/7244
 * https://github.com/grpc/grpc-go/blob/master/Documentation/anti-patterns.md#how-to-properly-create-a-clientconn-grpcnewclient

williammartin replied:
Best I can tell the commands that use this still work:

```
➜  sigstore-go git:(main) ~/workspace/cli/bin/gh cs ssh
? Choose codespace: github/github (wm/remove-device_authorization_grant_replica_race): ubiquitous capybara
Welcome to Ubuntu 20.04.6 LTS (GNU/Linux 6.5.0-1022-azure x86_64)

 * Documentation:  https://help.ubuntu.com
 * Management:     https://landscape.canonical.com
 * Support:        https://ubuntu.com/pro
Last login: Thu Jul 25 05:34:25 2024 from ::1
```

```
➜  sigstore-go git:(main) ~/workspace/cli/bin/gh cs jupyter
? Choose codespace: ipdor/StudyAtCoursera (master): organic space enigma
http://localhost:53733/?token=27038fd3fb270ee7e4fd4d6710c5d21de389c03539091fd8
```

williammartin commented:
@cli/codespaces can you double check that there's nothing surprising about the `invoker.go` change and approve if you're happy?
